plutus-core/untyped-plutus-core/Language/UntypedPlutusCore/Core/Type.hs

```diff
@@ -92,6 +108,23 @@ termAnn (Delay ann _)    = ann
 termAnn (Force ann _)    = ann
 termAnn (Error ann)      = ann
 
+bindFun
+    :: (ann -> fun -> Term name uni fun' ann)
+    -> Term name uni fun ann
+    -> Term name uni fun' ann
+bindFun f = go where
```

Comment:
This looks a little strange because it could in principle convert a `Builtin` into a `LamAbs` or something.  In fact it's only called with an `f` which actually returns a `Builtin`, so that's OK.  Why not just combine `bindFun` and `mapFun` into a single function?  (I see that `bindFun` is exported, but I don't think it's used anywhere.  Is there a possibility that it might be needed elsewhere in the future?)

Reply:
... or maybe it's needed by the HOAS machine?

Reply:
Exactly. We store partial application in `fun` to restore them in case they never get saturated.